### PR TITLE
Make style check CI clang-format installation more robust

### PR DESCRIPTION
**Pull request by dominiklohmann**
This started failing recently. We used Linuxbrew to install clang-format historically in the Style Check CI workflow because it was (1) pre-installed and (2) allowed for installing a more recent clang-format version. (1) looks to no longer be the case on newer runners, and (2) is also not true anymore, so let's just install it from PyPI.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [x] All user-facing changes have changelog entries
- [x] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

File: .github/workflows/style-check.yaml [hunk 1]
```diff
@@ -18,8 +18,9 @@ jobs:
 
       - name: Clang Format
         run: |
-          brew install clang-format
-          PATH="$(brew --prefix clang-format)/bin:${PATH}"
+          pip install --upgrade clang-format
+          which clang-format
```
Comment (mavam) [inline .github/workflows/style-check.yaml:22]: Why do we need this?

File: .github/workflows/style-check.yaml [hunk 2]
```diff
@@ -18,8 +18,9 @@ jobs:
...
+          clang-format --version
```
Comment (mavam) [inline .github/workflows/style-check.yaml:23]: Is this just diagnostics?

**Review by mavam — APPROVED** [review]

File: .github/workflows/style-check.yaml (hunk 1 above)
Comment (dominiklohmann) [reply .github/workflows/style-check.yaml:22]: If this job fails this and the clang-format version are the first things you're looking for if you cannot reproduce the failure locally.

File: .github/workflows/style-check.yaml (hunk 2 above)
Comment (dominiklohmann) [reply .github/workflows/style-check.yaml:23]: See https://github.com/tenzir/vast/pull/2598#discussion_r983815089

**Pull request closed without merging**